api/server/router/network/filter.go

```diff
@@ -8,6 +8,26 @@ import (
 	"github.com/docker/docker/runconfig"
 )
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
+	case true:
+		for _, nw := range nws {
+			if len(nw.Containers) == 0 && len(nw.Services) == 0 && !runconfig.IsPreDefinedNetwork(nw.Name) {
+				retNws = append(retNws, nw)
+			}
+		}
+	case false:
+		for _, nw := range nws {
+			if len(nw.Containers) > 0 || len(nw.Services) > 0 || runconfig.IsPreDefinedNetwork(nw.Name) {
```

Comment:
I believe these fields are provided by the gossip layer and are eventually consistent. Is it a problem if the dangling filter may provide results that are slightly out of date?

Reply:
I don't think it should be a problem as long *as the consumer knows* this is potentially racey.
Trying to rm a network from this list wouldn't really pose a problem except for potentially racey behavior from the state propagation, but this problem would exist in any case.